### ISSUE TITLE
"Error: Could not find a suitable provider for foreman_smartproxy"

### ISSUE THREAD

**Issue by threepistons** [theforeman/puppet-foreman_proxy#536]
Given:
- Puppet 6.9.0
- Puppetserver 6.6.0
- CentOS 7.7.1908
- Control repo:
  - [Puppetfile](https://github.com/theforeman/puppet-foreman_proxy/files/3636771/Puppetfile.txt)
  - [manifests/site.pp](https://github.com/theforeman/puppet-foreman_proxy/files/3636780/site.pp.txt)
  - [hiera.yaml](https://github.com/theforeman/puppet-foreman_proxy/files/3636788/hiera.yaml.txt)
  - [data/common.yaml](https://github.com/theforeman/puppet-foreman_proxy/files/3636819/common.yaml.txt)
  - [data/os/RedHat/CentOS.yaml](https://github.com/theforeman/puppet-foreman_proxy/files/3636826/CentOS.yaml.txt)
  - [data/os/RedHat/CentOS/7.yaml](https://github.com/theforeman/puppet-foreman_proxy/files/3636828/7.yaml.txt)
  - [data/nodes/puppet.redacteddomain.yaml](https://github.com/theforeman/puppet-foreman_proxy/files/3636855/puppet.redacteddomain.yaml.txt)

When I: run `/opt/puppetlabs/puppet/bin/r10k deploy environment production --puppetfile ; puppet generate types --environment production ; puppet agent -t` on puppet.redacteddomain.

I get:
```          
Info: Using configured environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Retrieving locales
Info: Loading facts
Info: Caching catalog for puppet.redacteddomain
Info: Applying configuration version '1568994223'
Info: Stage[main]: Unscheduling all events on Stage[main]
Error: Could not find a suitable provider for foreman_smartproxy
Notice: Applied catalog in 20.74 seconds
```
I expected: an error-free puppet run.

**Comment by ekohl** [comment]
You do have the `oauth` gem installed?

**Comment by threepistons** [comment]
I have a RPM called "tfm-rubygem-oauth-0.5.4-3" installed.

**Comment by ekohl** [comment]
That's installed in the SCL but Puppet bundles its own Ruby. We do have a `puppet-agent-oauth` package which installs the gem there. Be aware that when you upgrade your Puppet's major version they may change it (like from Puppet 5 to Puppet 6) and you need to reinstall it.

**Comment by threepistons** [comment]
Ok, the puppet-agent-auth package was installed and the latest version and from your repo, yet `yum reinstall puppet-agent-oauth` fixed the problem.  Thank you!

**Comment by misse** [comment]
Fixed my issue as well, ubuntu 18.04 foreman 2.0. I ran `sudo apt install --reinstall puppet-agent-oauth` and it solved the issue for me

**Comment by ekohl** [comment]
For the installer we have https://github.com/theforeman/foreman-installer/blob/develop/hooks/pre_validations/31-puppet-agent-oauth.rb but right now it only knows about RPM. A patch to add Debian support would be welcome. That would fix it automatically (for the installer).
